code/game/machinery/doors/airlock.dm

```diff
@@ -500,6 +501,81 @@ About the new airlock wires panel:
 				sleep(6)
 				update_icon(AIRLOCK_CLOSED)
 
+
+/obj/machinery/door/airlock/proc/change_paintjob(obj/item/airlock_painter/W, mob/user)
+	if(!paintable)
+		to_chat(user, "<span class='warning'>You cannot paint this type of airlock.</span>")
+		return FALSE
+
+	var/list/optionlist
+	if(airlock_material == "glass")
+		optionlist = list("Standard", "Public", "Engineering", "Atmospherics", "Security", "Command", "Medical", "Research", "Science", "Mining", "Maintenance", "External", "External Maintenance")
+	else
+		optionlist = list("Standard", "Public", "Engineering", "Atmospherics", "Security", "Command", "Medical", "Research", "Freezer", "Science", "Mining", "Maintenance", "External", "External Maintenance")
+
+	var/paintjob = input(user, "Please select a paintjob for this airlock.") in sortList(optionlist)
+	if((!in_range(src, user) && loc != user) || !W.paint(user))
+		return
+	switch(paintjob)
+		if("Standard")
+			icon = 'icons/obj/doors/airlocks/station/public.dmi'
+			overlays_file = 'icons/obj/doors/airlocks/station/overlays.dmi'
+			assemblytype = /obj/structure/door_assembly
+		if("Public")
+			icon = 'icons/obj/doors/airlocks/station2/glass.dmi'
+			overlays_file = 'icons/obj/doors/airlocks/station2/overlays.dmi'
+			assemblytype = /obj/structure/door_assembly/door_assembly_public
```

Comment:
Instead of copy-pasting these three values from `/obj/machinery/door/airlock/public`, maybe it's better to create a `/obj/machinery/door/airlock/public` value here and copy the values from it?
Something like (pseudocode, probably doesn't compile):
```js
var/obj/machinery/door/airlock/painted
switch(paintjob)
  if("Standard")
    painted = new/obj/machinery/door/airlock()
  if("Public")
    painted = new/obj/machinery/door/airlock/public()
  ...
icon = painted.icon
overlays_file = painted.overlays_file
assemblytype = painted.assemblytype
```

You can then even replace the switch with a assoc list of `name -> desired airlock type path` to make it even cleaner.

Reply:
I'm honestly not sure how to do what you're suggesting with the assoc lists, but you're first suggestion is definitely an improvement. I'll go with that for now.

Reply:
> I'm honestly not sure how to do what you're suggesting with the assoc lists,

Once you have 
```js
var/obj/machinery/door/airlock/painted
switch(paintjob)
  if("Standard")
    painted = new/obj/machinery/door/airlock()
  if("Public")
    painted = new/obj/machinery/door/airlock/public()
  if("Engineering")
    painted = new/obj/machinery/door/airlock/door_assembly_eng()
  ...
icon = painted.icon
overlays_file = painted.overlays_file
assemblytype = painted.assemblytype
```
you can easily transform that to 
```js
var/list/airlocks = new list(
  "Standard" = /obj/machinery/door/airlock,
  "Public" = /obj/machinery/door/airlock/public,
  "Engineering" = /obj/machinery/door/airlock/door_assembly_eng,
  ...
)
var/chosen_path = airlocks[paintjob]
var/obj/machinery/door/airlock/painted = new chosen_path ()

icon = painted.icon
overlays_file = painted.overlays_file
assemblytype = painted.assemblytype
```

And once you have that, it's easy to generate the `var/list/airlocks` by iterating over existing types with `typesof` instead of hardcoding the `name -> type` correspondence

Reply:
The issue is that all the names like "Standard", "Engineering", "Public" etc. are custom names. Those aren't actually the names of those types of airlocks, so I'm going to have to hardcode the list of names/paths for all the paintable airlocks. But even hardcoding them in, assoc lists are way better than what I have currently.

Reply:
You can get the non-hardcoded names roughly this way:
```js
var/airlocks = new list()
for(var/airlock_path in typesof(/obj/machinery/door/airlock))
  var/obj/machinery/door/airlock/A = new airlock_path()
  var/obj/structure/door_assembly/AT = A.assemblytype
  var/door_name = AT.base_name
  airlocks[door_name] = airlock_path
```
these `door_name`s would be slightly different from the ones you have currently, but that's probably not a big deal.

You can be fancy and instead of generating this list each time you want to paint a door, you can store it in a new var, or in a datum of its own.
But I'm not sure it's worth the effort.
(I'm also not sure of the performance implication of the `typesof()` search above. I *think* it shouldn't be a problem at all, but I'm not sure. If it is - then hardcoding the options might be better)